go-controller/cmd/ovnkube-trace/ovnkube-trace.go

```diff
@@ -462,17 +479,22 @@ func getPodInfo(coreclient *corev1client.CoreV1Client, restconfig *rest.Config,
 		return nil, err
 	}
 
+	podInfo, err = getDatabaseURIs(coreclient, restconfig, ovnNamespace, podInfo)
+	if err != nil {
+		klog.Exitf("Failed to get database URIs: %v\n", err)
+	}
+
 	// Find rtos MAC (this is the pod's first hop router).
-	lspCmd := "ovn-sbctl --no-leader-only " + cmd + " --bare --no-heading --column=mac list Port_Binding " + types.RouterToSwitchPrefix + podInfo.NodeName
-	ipOutput, ipError, err := execInPod(coreclient, restconfig, ovnNamespace, podInfo.OvnKubePodName, "ovnkube-node", lspCmd, "")
+	podInfo.RtosMAC, err = getSwitchPortMacAddress(coreclient, restconfig, podInfo, ovnNamespace, types.RouterToSwitchPrefix)
 	if err != nil {
-		return nil, fmt.Errorf("execInPod() failed. err: %s, stderr: %s, stdout: %s, podInfo: %v", err, ipError, ipOutput, podInfo)
+		return nil, err
 	}
-	macIP := strings.Split(strings.Replace(ipOutput, "\n", "", -1), " ")
-	if len(macIP) != 2 {
-		return nil, fmt.Errorf("invalid output %s", ipOutput)
+
+	// Find rtots MAC (this is the pod's first hop router when ovn is in interconnected zone).
```

Comment:
what about non-interconnect? how does trace work in nonIC if we pass the `types.RouterToTransitSwitchPrefix` ? we use the above one i.e ` getSwitchPortMacAddress` ? So if its single-zone OR nonIC we use `RtosMAC` and if its multizone we use `RtotsMAC`  if pods are on different nodes and if not its the simple same switch trace huh?

Reply:
The `RtotsMAC` is used only for IC case for collecting ovn trace on the remote node where remote node is on different zone, This mac would be just empty for non-IC case and never get used.

Reply:
So flow is `podA -> nodeA switch -> nodeA router -> transitswitch -> nodeB router -> nodeB switch -> podB` for IC: so here the next hop for the ovn-trace from podA is the `RtotsMAC` and not `RtosMAC`?

Reply:
yes @tssurya , that's correct. when packet arrives on the nodeB, it would have `RtotsMAC` as the destination MAC.